mittab/libs/outround_tab_logic/pairing.py

```diff
@@ -0,0 +1,112 @@
+import math
+
+from mittab.apps.tab.models import *
+
+from mittab.libs.outround_tab_logic.checks import have_enough_rooms
+from mittab.libs.outround_tab_logic.bracket_generation import gen_bracket
+
+from mittab.libs.tab_logic import have_properly_entered_data
+from mittab.libs import errors
+
+
+def good_to_go(num_teams):
+    if num_teams > Team.objects.count():
+        raise errors.NotEnoughTeamsError()
+
+    if num_teams < 2:
+        raise errors.NotEnoughTeamsError()
+
+    # Check there are enough rooms
+    if not have_enough_rooms()[0]:
+        raise errors.NotEnoughRoomsError()
+
+    # If we have results, they should be entered and there should be no
+    # byes or noshows for teams that debated
+    round_to_check = TabSettings.get("tot_rounds", 5)
+    have_properly_entered_data(round_to_check)
+
+
+def get_next_available_room(num_teams, type_of_break):
+    base_queryset = Outround.objects.filter(num_teams=num_teams,
+                                            type_of_round=type_of_break)
+
+    var_to_nov = TabSettings.get("var_to_nov", 2)
+
+    other_queryset = Outround.objects.filter(type_of_round=not type_of_break)
+
+    if type_of_break == BreakingTeam.VARSITY:
+        other_queryset = other_queryset.filter(num_teams=num_teams / var_to_nov)
+    else:
+        other_queryset = other_queryset.filter(num_teams=num_teams * var_to_nov)
+
+    rooms = [r.room
+             for r in RoomCheckIn.objects.filter(round_number=0) \
+             .prefetch_related("room")]
+    rooms.sort(key=lambda r: r.rank, reverse=True)
+
+    for room in rooms:
+        if not base_queryset.filter(room=room).exists() and \
+           not other_queryset.filter(room=room).exists():
+            return room
+    return None
+
+
+def gov_team(team_one, team_two):
```

Comment:
This needs to be fixed, right?

Reply:
So here's my thoughts on this:

There's a bunch of logic here with regard to the method that different tournaments do outrounds (flip for choice, least choice chooses, flip for side, sidelocked or not, etc).  I think the best way to implement this is through at least a `SIDELOCK` `TabSetting`.  There should then be a toggle for each outround that is like are these sides real, or are they placeholders to be changed after coin flip, etc.  Then the display will change.  Been thinking about the best way to do that (I don't plan on having it keep track of choice, etc).

Thoughts ?

Reply:
I see. This seems like a reasonable placeholder.